study.md

```diff
@@ -246,5 +292,15 @@ Pick a few design heuristics. Describe some common, named refactorings you can
 use approach a good design.
 
 ```md
-<!-- your answer here -->
+-Recognizing code smells and fixing them.  Code smells are indications that there
+may be a deeper problem in the source code.  They indicate violations of
+fundamental design principles.
+-Move work into private methods
+-Make sure a class doesn't have multiple responsibilities
+-Be aware of large conditional statements
+-Avoid data clumps such as arrays of arrays (array[i][j])
+-Use semantic naming, so that it is easy to understand what your code does
+
+Sources: [Code Smells - Wikipedia](https://en.wikipedia.org/wiki/Code_smell)
+[Code Smells: Your Refactoring Cheat Codes](http://blog.endpoint.com/2013/04/code-smells-your-refactoring-cheat-codes.html)
 ```
```

Comment:
:+1: Yep yep! Code retreat illustrated many of these.

What would you use to DRY up duplicated logic? Maybe "Extract Method". Have a look at some named refactorings, like "Extract Method", to level up.